types/index.d.ts

```diff
@@ -162,36 +162,36 @@ declare namespace math {
   }
 
   interface NodeCtor {
-    new (): MathNode
+    new (): BaseNode
   }
 
-  interface AccessorNode<TObject extends MathNode = MathNode> extends MathNode {
+  interface AccessorNode<TObject extends BaseNode = BaseNode> extends BaseNode {
     type: 'AccessorNode'
     isAccessorNode: true
-    object: TObject
+    object: BaseNode
     index: IndexNode
     name: string
   }
   interface AccessorNodeCtor {
-    new <TObject extends MathNode = MathNode>(
+    new <TObject extends BaseNode = BaseNode>(
       object: TObject,
       index: IndexNode
     ): AccessorNode<TObject>
   }
 
-  interface ArrayNode<TItems extends MathNode[] = MathNode[]> extends MathNode {
+  interface ArrayNode<TItems extends BaseNode[] = BaseNode[]> extends BaseNode 
```

Comment:
I think there may have been an error resolving a merge conflict or something... There's a `{` missing here which is resulting in some type errors.